Azure/azure-cli
# [Packaging] Fix #22741: `az upgrade`: This command becomes non-blocking on Windows

**Description**

Close https://github.com/Azure/azure-cli/issues/22741, close https://github.com/Azure/azure-cli/issues/23947

This PR changes the behavior of `az upgrade` on Windows.
If the Python version changes in new version, the upgrade is blocked.
(This issue can be reproduced by installing 2.44 and running `az upgrade`)
![image](https://github.com/Azure/azure-cli/assets/2227874/4f2e80e1-f0c8-4814-88bc-34f732a201e8)


In this PR, `az upgrade` exits after MSI installation dialog is opened. So the related file is not opened during upgrade.

Three breaking changes:
1. `az upgrade` becomes non-blocking command, it exit after MSI installation dialog is opened.
2. az does not know if the upgrade is successful, the exit code is always 0 once installation starts.
3. This command does not upgrade extensions if new version is available. Users need to run `az upgrade` again to upgrade extensions.


**Testing Guide**

set this in `upgrade_version`
```python
local_version = '2.50.0'
installer = 'MSI'
```


---

This checklist is used to make sure that common guidelines for a pull request are followed.
